src/nvim/main.c

```diff
@@ -600,6 +588,12 @@ int main(int argc, char **argv)
     TIME_MSG("UIEnter autocommands");
   }
 
+#ifdef MSWIN
+  if (use_builtin_ui) {
+    os_icon_init();
```

Comment:
@justinmk With `use_builtin_ui`, the icon isn't being set, as of `v0.9.0-dev-823+g41aa5ce3eb49`.  I haven't bisected.

Reply:
It's probably because of the PR where TUI was split from the main nvim process.  We could remove this `if` check or OR `use_remote_ui` with `use_builtin_ui`.  I don't know which is better.

Reply:
er... `use_builtin_ui` should be true, why isn't it?

Reply:
I ran these commands:
```
$ gdb .\build\bin\nvim.exe
(gdb) b main.c:303
(gdb) r
(gdb) n
(gdb) p use_builtin_ui
$1 = 0x1
```

Note that the value of `use_builtin_ui` is true.  But I noticed that this nvim process, the one gdb is attached to, runs into this line, and after executing it, the nvim window appears.  At that point, I don't have gdb's ui anymore.

https://github.com/neovim/neovim/blob/964ae205a583807127eda75d9876fb6dfda6065c/src/nvim/main.c#L372-L373

---

I don't understand how the two nvim processes work.  I asked in nvim-dev matrix channel about where the embedded process is created, but at that moment no one was able to provide an answer.  What I've written above is only the little info I was able to gather; can't say if it's helpful.

Reply:
Nope.  Doesn't satisfy `if(ui_client_channel_id)` either.

Reply:
I think it should either be `if(use_builtin_ui || embedded_mode)` or just `if(embedded_mode)`.  But I'm struggling with understanding how the main and embedded processes are being run.  Can't tell what the conditional should be.

Reply:
> I think it should either be `if(use_builtin_ui || use_remote_ui)`

👍  that seems fine.